lib/matplotlib/type1font.py

```diff
@@ -265,10 +268,9 @@ def fontmatrix(array):
             newmatrix = np.dot(modifier, oldmatrix)
             array[::2] = newmatrix[0:3, 0]
             array[1::2] = newmatrix[0:3, 1]
-            # Not directly using `b'%a' % x for x in array` for now as that
-            # produces longer reprs on numpy<1.14, causing test failures.
-            as_string = '[' + ' '.join(str(x) for x in array) + ']'
-            return as_string.encode('latin-1')
+            return (
+                '[' + ' '.join(f'{x:.5g}' for x in array) + ']'
```

Comment:
perhaps .5f instead?

Reply:
I cannot comment on g vs. f, but I would prefer
```suggestion
                '[%s]' % ' '.join(f'{x:.5g}' for x in array)
```
which is slightly more readable IMHO. But I won't block over this.

Reply:
(the problem with 'g' is exponent notation: does postscript actually support notation such as '1e4'?)

Reply:
Postscript supports exponent notation:

> NumbersNumbers in the PostScript language include signed integers, such as
>   123  –98  43445  0  +17
> reals, such as
>   –.002  34.5  –3.62  123.6e10  1E–5  –1.  0.0
> and radix numbers, such as
>   8#1777  16#FFFE  2#1000

https://www.adobe.com/content/dam/acom/en/devnet/actionscript/articles/psrefman.pdf page 27

Reply:
Ah, that's fine then.

Reply:
Yeah, I thought we should keep a small number of significant digits instead of just rounding to a fixed number of decimals. But on second thought there is little point, since the scale is fixed: fonts are written in terms of 1000 or 1024 units as the side of the design square, so the values on the diagonal are around 0.001. The off-diagonal values are either zero or of similar magnitude as 0.001, otherwise there is no visible effect. We can as well use %f.